### ISSUE TITLE
Duplicate includes causes load errors in MaterialX

### ISSUE THREAD

**Issue by laserallan** [materialx/MaterialX#248]
It seems like including the same document twice in a materialX document causes an error related to duplicate nodes. It happens when documents are recursively included so any type of use where you include a document when a node is needed is likely to break if you have any type of hierarchical includes.
It can easily be solved by using CopyOptions to skip duplicate elements but using it locally is likely to produce documents that are incompatible with applications not using that setting. 

Here is a python script triggering the issue:
```
import MaterialX as mx

a = mx.createDocument()
a.addNodeDef('banana')
b = mx.createDocument()
b.importLibrary(a)
c = mx.createDocument()
c.importLibrary(a)
c.importLibrary(b)
```
Ideally I would like to be able to include any document from which I directly use nodes in my document regardless if these documents are dependent on each other and have it work in any materialx application regardless of how they set up their CopyOptions.

**Comment by jstone-lucasfilm** [comment]
We've discussed this a bit off-line, and one proposed solution is to automatically exclude elements in Document::importLibrary when both their name and _source URI_ match those of an element in the destination document.  That should handle any dependency that ends up included in a content document more than once, without diminishing the existing error handling for name collisions.

**Comment by jstone-lucasfilm** [comment]
Proposed fix in pull request #255

**Comment by jstone-lucasfilm** [comment]
Merged to master
